src/System.Runtime.Extensions/tests/System/AppDomainTests.cs

```diff
@@ -508,59 +627,68 @@ public void AssemblyResolve_RequestingAssembly()
 
-            Type t;
-            try
-            {
-                t = Type.GetType("Program", true);
-            }
-            finally
-            {
-                AppDomain.CurrentDomain.TypeResolve -= handler;
-            }
-            Assert.NotNull(t);
+                Type t;
+                try
+                {
+                    t = Type.GetType("Program", true);
+                }
+                finally
+                {
+                    AppDomain.CurrentDomain.TypeResolve -= handler;
+                }
+                Assert.NotNull(t);
+                return SuccessExitCode;
+            }).Dispose();
         }
 
         [Fact]
         public void ResourceResolve()
         {
-            ResourceManager res = new ResourceManager(typeof(FxResources.TestApp.SR));
-            Assert.Throws<MissingManifestResourceException>(() => res.GetString("Message"));
+            RemoteInvoke(() => {
+                ResourceManager res = new ResourceManager(typeof(FxResources.TestApp.SR));
+                Assert.Throws<MissingManifestResourceException>(() => res.GetString("Message"));
 
-            ResolveEventHandler handler = (sender, args) =>
-            {
-                return Assembly.Load("TestApp");
-            };
+                ResolveEventHandler handler = (sender, args) =>
+                {
+                    return Assembly.Load("TestApp");
+                };
 
-            AppDomain.CurrentDomain.ResourceResolve += handler;
+                AppDomain.CurrentDomain.ResourceResolve += handler;
 
-            String s;
-            try
-            {
-                s = res.GetString("Message");
-            }
-            finally
-            {
-                AppDomain.CurrentDomain.ResourceResolve -= handler;
-            }
-            Assert.Equal(s, "Happy Halloween");
+                String s;
+                try
+                {
+                    s = res.GetString("Message");
+                }
+                finally
+                {
+                    AppDomain.CurrentDomain.ResourceResolve -= handler;
+                }
+                Assert.Equal(s, "Happy Halloween");
+                return SuccessExitCode;
+            }).Dispose();
         }
 
         [Fact]       
         public void SetThreadPrincipal()
         {
-            Assert.Throws<ArgumentNullException>(() => {AppDomain.CurrentDomain.SetThreadPrincipal(null);});
-            var identity = new System.Security.Principal.GenericIdentity("NewUser");
-            var principal = new System.Security.Principal.GenericPrincipal(identity, null);
-            AppDomain.CurrentDomain.SetThreadPrincipal(principal);
+            RemoteInvoke(() => {
+                Assert.Throws<ArgumentNullException>(() => { AppDomain.CurrentDomain.SetThreadPrincipal(null); });
+                var identity = new System.Security.Principal.GenericIdentity("NewUser");
```

Comment:
coding guidelines say 'var' is fine when it's obvious what the type is eg when it's assigned 'new' like this.